pysal/contrib/viz/color.py

```diff
@@ -0,0 +1,178 @@
+                      colormap scale type  [sequential, diverging, qualitative]
+
+    k               : int
+                      number of classes
+
+    color_encoding  : string
+                      encoding of colors [hexc, rgb, mpl, mpl_colormap]
+                      * hex: list of hex strings
+                      * rgb: list of RGB 0-255 triplets
+                      * mpl: list of RGB 0-1 triplets as used by matplotlib
+                      * mpl_colormap: matplotlib color map
+
+    Returns
+    -------
+
+    colors:  color map in the specified color_encoding
+
+    """
+    encs = {'hex': 'hex_colors',
+            'rgb': 'colors',
+            'mpl': 'mpl_colors',
+            'mpl_colormap': 'mpl_colormap'}
+    if not palette:
+        cmtype = pltt2type[name.lower()]
+        if name[-2:] == '_r':
+            palette = pltt.colorbrewer.get_map(name[:-2], cmtype,
+                    k, reverse=True)
+        else:
+            palette = pltt.colorbrewer.get_map(name, cmtype, k)
+    colors = getattr(palette, encs[color_encoding.lower()])
+    return colors
+
+def _build_pltt2type():
+    types = ['sequential', 'diverging', 'qualitative']
+    pltt2type = {}
+    for t in types:
+        pals = list(set([
+            p.split('_')[0] for p in dir(getattr(pltt.colorbrewer, t)) 
+                            if p[0]!='_'
+                            ]))
+        pals = pals + [p+'_r' for p in pals]
+        for p in pals:
+            pltt2type[p.lower()] = t
+    return pltt2type
+
+pltt2type = _build_pltt2type()
+
+def get_color_map_b2m(name='BuGn', cmtype='sequential', k=5,
+                  color_encoding='hexc'):
+
+    """
+    DEPRECATED!!!
+
+    Get a brewer colormap from `brewer2mpl`
+
+    Arguments
+    ---------
+
+    name:   string
+            colormap name
+
+    cmtype: string
+            colormap scale type  [sequential, diverging, qualitative]
+
+    k:  int
+        number of classes
+
+    color_encoding: string
+                    encoding of colors [hexc, rgb, mpl, mpl_colormap]
+                    hex: list of hex strings
+                    rgb: list of RGB 0-255 triplets
+                    mpl: list of RGB 0-1 triplets as used by matplotlib
+                    mpl_colormap: matplotlib color map
+
+    Returns
+
+    colors:  color map in the specified color_encoding
+
+    """
+    encs = {'hexc': 'hex_colors',
+            'rgb': 'colors',
+            'mpl': 'mpl_colors',
+            'mpl_colormap': 'mpl_colormap'}
+    try:
+        bmap = brewer2mpl.get_map(name, cmtype, k)
+        colors = getattr(bmap, encs[color_encoding.lower()])
+        return colors
+    except:
+        print('Color map not found: ', name, cmtype, k)
+
+def get_maps_by_type(data_type):
+    names = [name for name in dir(data_type) if not name.startswith('_')]
+    return names
+
+# get maps for each ctype and default k=5 for populating display options
+ctypes = (b2m_sequential, b2m_diverging, b2m_qualitative)
```

Comment:
I can see two ways to go.

 A) we dump brewer2mpl (which I thought was the plan) and rewrite line 141 to do the right thing with whatever replaced brewer2mpl. 

B) Lines 133-178 were really helpers for ChoroBrewer, so they could get deleted from color.py and reimplemented in ChoroBrewer using the new api in color.py. I'm not sure those lines are used elsewhere?

I think B is preferred as brewer2mpl is deprecated so keeping it around as even a soft dependency doesn't smell good. But, if we go this B route, the api in color.py should support the a user being able to build something like lines 133-178 when importing from color.py